### ISSUE TITLE
[FEAT][MANIFEST] Add reduction and scan ops to ops_manifest.yaml

### ISSUE THREAD

**Issue by lcy-seso** [tile-ai/TileOPs#744]
Parent: #741
Depends: #743 (manifest.md spec revision must land first)

### Symptom / Motivation

`ops_manifest.yaml` has **zero reduction/scan ops** despite ~20 implemented in code. This blocks spec-driven validation, benchmark generation, and nightly reporting for these op families.

### Root Cause Analysis

Reduction and scan ops were implemented before the manifest system was established. They were never backfilled into the manifest.

### Related Files

- `tileops/ops_manifest.yaml`
- `tileops/kernels/reduction/`
- `tileops/ops/reduction/`
- `tests/ops/test_softmax.py`
- `tests/ops/test_reduce.py`
- `benchmarks/ops/bench_softmax.py`
- `benchmarks/ops/bench_reduce.py`

## Goal

Add manifest entries for all ~20 reduction ops and 2 scan ops, conforming to the revised spec from #743 (ordered dict params, `status` field, roofline `vars`, `keepdim`, PyTorch-aligned interfaces).

### Step 1: Add softmax family (`family: reduction`)

| Op name | Op class | Notes |
|---|---|---|
| `softmax_fwd` | `SoftmaxOp` | output shape == input shape, no `keepdim` |
| `log_softmax_fwd` | `LogSoftmaxOp` | output shape == input shape, no `keepdim` |
| `logsumexp_fwd` | `LogSumExpOp` | output shape == input shape, no `keepdim` |

Entry format:

```yaml
softmax_fwd:
  family: reduction
  status: implemented

  signature:
    inputs:
      x: {dtype: "float16 | bfloat16"}
    outputs:
      y: {dtype: "same_as(x)"}
    params:
      dim: {type: int, default: -1}
    shape_rules:
      - "y.shape == x.shape"

  workloads:
    - {x_shape: [32, 32, 4096], dtypes: [float16, bfloat16], label: "attn-weights-4k"}
    - {x_shape: [32, 32, 32768], dtypes: [bfloat16], label: "attn-weights-32k"}
    - {x_shape: [4, 102400], dtypes: [float16, bfloat16], label: "lm-head-logits"}

  roofline:
    vars:
      M: "product(x.shape[:dim])"
      N: "x.shape[dim]"
    flops: "5 * M * N"
    bytes: "2 * M * N * elem_bytes"

  source:
    kernel: tileops/kernels/reduction/softmax.py
    op: tileops/ops/reduction/softmax.py
    test: tests/ops/test_softmax.py
    bench: benchmarks/ops/bench_softmax.py
```

### Step 2: Add simple reductions (`family: reduction`)

| Op name | Op class | pad_value |
|---|---|---|
| `sum_fwd` | `SumOp` | 0.0 |
| `mean_fwd` | `MeanOp` | 0.0 |
| `amax_fwd` | `AmaxOp` | -inf |
| `amin_fwd` | `AminOp` | +inf |
| `prod_fwd` | `ProdOp` | 1.0 |

All simple reductions share this signature pattern (PyTorch-aligned: `dim` + `keepdim`):

```yaml
sum_fwd:
  family: reduction
  status: implemented

  signature:
    inputs:
      x: {dtype: "float16 | bfloat16 | float32"}
    outputs:
      y: {dtype: "same_as(x)"}
    params:
      dim: {type: "int | list[int]"}
      keepdim: {type: bool, default: false}
    shape_rules:
      - "y.ndim == x.ndim if keepdim else x.ndim - len(dim)"
      - "y.shape[i] == 1 if i in dim and keepdim else x.shape[i]"

  workloads:
    - {x_shape: [2048, 4096], dtypes: [float16, bfloat16], label: "hidden-state-reduce"}
    - {x_shape: [64, 32768], dtypes: [bfloat16], label: "long-seq-reduce"}

  roofline:
    vars:
      M: "product(x.shape[i] for i in range(x.ndim) if i not in dim)"
      N: "product(x.shape[i] for i in dim)"
    flops: "M * N"
    bytes: "(M * N + M) * elem_bytes"

  source:
    kernel: tileops/kernels/reduction/reduce.py
    op: tileops/ops/reduction/reduce.py
    test: tests/ops/test_reduce.py
    bench: benchmarks/ops/bench_reduce.py
```

### Step 3: Add Welford reductions (`family: reduction`)

| Op name | Op class | Extra params |
|---|---|---|
| `std_fwd` | `StdOp` | `correction: int` (default 1) |
| `var_fwd` | `VarOp` | `correction: int` (default 1) |
| `var_mean_fwd` | `VarMeanOp` | `correction: int` (default 1) |

PyTorch-aligned: `torch.var(input, dim, correction, keepdim)`.

### Step 4: Add argmax/argmin, logical, and norm reductions (`family: reduction`)

| Op name | Output dtype | `keepdim` |
|---|---|---|
| `argmax_fwd` | int64 | yes |
| `argmin_fwd` | int64 | yes |
| `all_fwd` | bool | yes |
| `any_fwd` | bool | yes |
| `count_nonzero_fwd` | int64 | yes |
| `l1_norm_fwd` | same_as(x) | yes |
| `l2_norm_fwd` | same_as(x) | yes |
| `inf_norm_fwd` | same_as(x) | yes |

### Step 5: Add scan ops (`family: scan`)

| Op name | Op class | Notes |
|---|---|---|
| `cumsum_fwd` | `CumsumOp` | output shape == input shape, no `keepdim` |
| `cumprod_fwd` | `CumprodOp` | output shape == input shape, no `keepdim` |

Scan ops use `family: scan` (not `reduction`). Output shape always equals input shape — no dimension is collapsed.

```yaml
cumsum_fwd:
  family: scan
  status: implemented

  signature:
    inputs:
      x: {dtype: "float16 | bfloat16 | float32"}
    outputs:
      y: {dtype: "same_as(x)"}
    params:
      dim: {type: int, default: -1}
    shape_rules:
      - "y.shape == x.shape"

  workloads:
    - {x_shape: [2048, 4096], dtypes: [float16, bfloat16], label: "hidden-state-scan"}
    - {x_shape: [64, 32768], dtypes: [bfloat16], label: "long-seq-scan"}

  roofline:
    vars:
      M: "product(x.shape[:dim])"
      N: "x.shape[dim]"
    flops: "M * N"
    bytes: "2 * M * N * elem_bytes"

  source:
    kernel: tileops/kernels/reduction/scan.py
    op: tileops/ops/reduction/scan.py
    test: tests/ops/test_scan.py
    bench: benchmarks/ops/bench_scan.py
```

### Step 6: Verify all entries

- `python -c "import yaml; yaml.safe_load(open('tileops/ops_manifest.yaml'))"` passes
- All `status: implemented` entries have existing source paths
- All `status: spec` entries have intended source paths (no file-existence check)

## Constraints

- No changes to Op implementation code (manifest-only PR)
- All entries must conform to revised manifest.md spec from #743
- All param names and semantics must match PyTorch's public API (R15)
- Workload shapes must reflect real LLM/CV workloads, not arbitrary numbers

## Acceptance Criteria

- [ ] AC-1: All 3 softmax-family ops have entries with `family: reduction`, roofline `vars`, no `keepdim`
- [ ] AC-2: All 5 simple reduction ops have entries with `dim` + `keepdim` params and conditional shape_rules
- [ ] AC-3: All 3 Welford ops have entries with `correction` + `dim` + `keepdim` params
- [ ] AC-4: All 8 other reduction ops (argmax, argmin, all, any, count_nonzero, l1/l2/inf_norm) have entries with correct output dtypes
- [ ] AC-5: `cumsum_fwd` and `cumprod_fwd` use `family: scan`, no `keepdim`
- [ ] AC-6: Every entry has `status` field (`spec` or `implemented`)
- [ ] AC-7: Every entry has roofline `vars` binding (per R14 from #743)
- [ ] AC-8: Source paths correct for all `status: implemented` entries
- [ ] AC-9: `python -c "import yaml; yaml.safe_load(open('tileops/ops_manifest.yaml'))"` passes
- [ ] AC-10: Modified files pass existing tests
